test/e2e/runner/cli_provisioner.go

```diff
@@ -328,3 +329,34 @@ func (cli *CLIProvisioner) FetchOpenShiftInfraLogs(logPath string) error {
 
 	return kerrors.NewAggregate(errs)
 }
+
+// FetchOpenShiftMachineLogs returns logs for Openshift machine components.
+func (cli *CLIProvisioner) FetchOpenShiftMachineLogs(cfg *config.Config, eng *engine.Engine, logPath string) error {
+	sshKeyPath := cfg.GetSSHKeyPath()
+
+	kubeConfig, err := kubernetes.GetConfig()
+	if err != nil {
+		return err
+	}
+	master := fmt.Sprintf("%s@%s", eng.ClusterDefinition.Properties.LinuxProfile.AdminUsername, kubeConfig.GetServerName())
+
+	services := []string{"etcd.service"}
```

Comment:
Soon this will run as a pod. Can you try to separate the logic based on the orchestrator version?

Reply:
Follow-up.